fennel.lua

```diff
@@ -2946,6 +2946,18 @@ Same as ->> except will short-circuit with nil when it encounters a nil value."
          (assert body1 "expected body")
          `(if ,condition
               (do ,body1 ,...)))
+ :let-open (fn [closable-bindings ...]
+              "Like `let`, but invokes (v:close) on every binding after evaluating the body.
+The body is evaluated inside `xpcall` so that bound values will be closed upon
+encountering an error before propagating it."
+              (let [bodyfn    `(fn [] ,...)
+                    closer    `(fn close-handlers# [ok# ...] (if ok# ... (error ... 0)))
```

Comment:
The use of `...` here is a little funny; doesn't the fact that `...` is used in the middle of the `if` and `error` calls mean that it will collapse down to a single value? Seems like we should use `val#` here instead maybe, or use `(values ...)` in the first branch of the `if`.

Reply:
So, this could either use a comment explainer or a more verbose change for better clarity, but the fact that it collapses down to a single value is actually why it behaves correctly. I originally had `(let [msg ...] (error msg 0))`, but `(error ... 0)` is semantically equivalent without the extra variable binding, so I typed it that way.

Definitely happy to change it, I just figured in the name of keeping emitted code as overhead-free as possible that this implementation makes the most sense; definitely comment-worthy, though.

Reply:
Oh! I wasn't thinking of the `error` part; I was assuming that the `(if ok# ...` part would collapse down to a single value. But since it's in a tail position I guess it does actually work? I assumed it wouldn't since it's not at the end of the form, but now that I think about it more, it should be the tail position that's more relevant here. Just tested it in the repl and this appears to be the case, so never mind!